src/Components/test/E2ETest/Tests/VirtualizationTest.cs

```diff
@@ -211,6 +212,33 @@ public void CanUseViewportAsContainer()
             Browser.NotEqual(expectedInitialSpacerStyle, () => topSpacer.GetAttribute("style"));
         }
 
+        [Fact]
+        public async Task ToleratesIncorrectItemSize()
+        {
+            Browser.MountTestComponent<VirtualizationComponent>();
+            var topSpacer = Browser.Exists(By.Id("incorrect-size-container")).FindElement(By.TagName("div"));
+            var expectedInitialSpacerStyle = "height: 0px;";
+
+            // Wait until items have been rendered.
+            Browser.True(() => GetItemCount() > 0);
+            Browser.Equal(expectedInitialSpacerStyle, () => topSpacer.GetAttribute("style"));
+
+            // Scroll slowly, in increments of 50px at a time. At one point this would trigger a bug
+            // due to the incorrect item size, whereby it would not realise it's necessary to show more
+            // items because the first time the spacer became visible, the size calculation said that
+            // we're already showing all the items we need to show.
+            for (var pos = 0; pos < 1000; pos += 50)
+            {
+                Browser.ExecuteJavaScript($"document.getElementById('incorrect-size-container').scrollTop = {pos};");
+                await Task.Delay(200);
```

Comment:
Is there a way to see if a specific item is visible instead? We've had trouble in the past where `Task.Delays` in the past where delays haven't been sufficient on slower CI machines.

Reply:
The delay isn't waiting for anything to happen in the UI, it's just representing the fact that as a user scrolls by dragging the mouse, it occurs as a series of steps. At some point in this series the bug occurs (if the fix wasn't in this PR), but it's not useful to specify at exactly what point.